checkov/main.py

```diff
@@ -139,6 +139,11 @@ def run(banner=checkov_banner, argv=sys.argv[1:]):
     external_checks_dir = get_external_checks_dir(config)
     url = None
     created_baseline_path = None
+
+    if sum(arg is not None for arg in [config.soft_fail, config.soft_fail_on, config.hard_fail_on]) > 1:
+        parser.error("'--soft-fail', '--soft-fail-on' and '--hard-fail-on' cannot be used together.")
+        return
```

Comment:
@kartikp10 Consider using `argparse`'s `add_mutually_exclusive_group` (inherited in configargparse) instead please 🙏

Reply:
Got it!